File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    const stack = [];
    for (let i of s){
        if (i == '('){
            stack.push(i)
        }else{
            if (stack.length >= 1 && stack[stack.length-1] == '('){
                stack.pop()
            }else{
                return false
            }
        }
    }
    if (stack.length == 0){
        return true
    }else{
        return false
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    stack = []
    if len(s) == 1 :
        return False
    
    for i in s:
        if i == '(':
            stack.append(i)
        else :
            if len(stack) >= 1 and stack[-1] == '(':
                stack.pop()
            else :
                return False
    if stack :
        return False
    else:
        return True